src/Workspaces/CSharp/Portable/Simplification/Reducers/AbstractCSharpReducer.AbstractReductionRewriter.cs

```diff
@@ -22,10 +21,10 @@ protected abstract class AbstractReductionRewriter : CSharpSyntaxRewriter, IRedu
         {
             private readonly ObjectPool<IReductionRewriter> _pool;
 
-            protected CSharpParseOptions ParseOptions { get; private set; }
-            protected OptionSet OptionSet { get; private set; }
+            protected CSharpParseOptions? ParseOptions { get; private set; }
+            protected CSharpSimplifierOptions? Options { get; private set; }
             protected CancellationToken CancellationToken { get; private set; }
-            protected SemanticModel SemanticModel { get; private set; }
+            protected SemanticModel? SemanticModel { get; private set; }
```

Comment:
these are odd nullable cases.  they're nullable because they get pooled.  but hte code that then uses the object should assume it's in an initialized state.  my personal preference it to make them non-nullable, but i don't feel super strongly.